### ISSUE TITLE
bug: elasticsearch DomainStatus.Created may stay false forever

### ISSUE THREAD

**Issue by thrau** [localstack/localstack#4686]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running `awslocal es create-elasticsearch-domain --domain-name foobar` for the first time, elasticsearch is downloaded, which takes a while...

Checking the health endpoint returns a healthy cluster:
```zsh
% curl -s http://localhost:4571/_cluster/health | jq .                                       
{
  "cluster_name": "elasticsearch",
  "status": "green",
  "timed_out": false,
  "number_of_nodes": 1,
  "number_of_data_nodes": 1,
  "active_primary_shards": 0,
  "active_shards": 0,
  "relocating_shards": 0,
  "initializing_shards": 0,
  "unassigned_shards": 0,
  "delayed_unassigned_shards": 0,
  "number_of_pending_tasks": 0,
  "number_of_in_flight_fetch": 0,
  "task_max_waiting_in_queue_millis": 0,
  "active_shards_percent_as_number": 100
}
```

But `awslocal es describe-elasticsearch-domain --domain-name foobar | jq ".DomainStatus.Created"` always returns false.

### Expected Behavior

`awslocal es describe-elasticsearch-domain --domain-name foobar | jq ".DomainStatus.Created"` should return `true` !

### How are you starting LocalStack?

With the `localstack` script

### Steps To Reproduce

`DNS_ADDRESS=0 bin/localstack start --host`

### Environment

```markdown
- OS: Linux Mint 20
- LocalStack: latest
```

### Anything else?

This function is the issue, which times out after 2 minutes (the default value of `CLUSTER_STARTUP_TIMEOUT`), however the installation takes longer than two minutes.

https://github.com/localstack/localstack/blob/69dd7992babb3528f3e472c991ef237f3af8e756/localstack/services/es/es_api.py#L74-L88

Basically we should implement event-driven lifecycle hooks that are called when the cluster is up and healthy, or up and unhealthy.

**Comment by alexrashed** [comment]
This issue has been fixed with #5341 (by deleting the timeout all together).
